cmd/orb-agent/main.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright © 2021 NAME HERE <EMAIL ADDRESS>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/ns1labs/orb/agent"
+)
+
+var (
+	cfgFiles      []string
+	defaultConfig = "/etc/orb/agent.yml"
+
+	rootCmd = &cobra.Command{
+		Use:   "orb-agent",
+		Short: "orb-agent connects to orb control plane",
+		Long:  "orb-agent connects to orb control plane",
+		Run:   Run,
+	}
+)
+
+func Run(cmd *cobra.Command, args []string) {
+	var config agent.Config
+	viper.Unmarshal(&config)
+	fmt.Printf("%+v\n", config)
+	s, err := agent.New(config)
+	cobra.CheckErr(err)
+	cobra.CheckErr(s.Start())
+}
+
+func Execute() {
+	cobra.CheckErr(rootCmd.Execute())
+}
+
+func init() {
+	cobra.OnInitialize(initConfig)
+	rootCmd.Flags().StringSliceVarP(&cfgFiles, "config", "c", []string{}, "Path to config files (can be specified multiple times)")
+}
+
+func mergeOrError(path string) {
+	v := viper.New()
+	v.SetConfigFile(path)
+	cobra.CheckErr(v.ReadInConfig())
+
+	var fZero float64
+
+	// check that version of config files are all matched up
+	if versionNumber1 := viper.GetFloat64("version"); versionNumber1 != fZero {
```

Comment:
`orb-agent` expects to see multiple config files on start so something like

`orb-agent -c conf1.yml conf2.yml` `mergeOrError` basically tries to merge the keys overwriting with the later config files on conflict. This checks that the version string of all the files presented match or it errors. If we like this functionality we can comment this and maybe add a quick test.